components/homme/src/share/gllfvremap_mod.F90

```diff
@@ -448,54 +442,28 @@ subroutine gfr_fv_phys_to_dyn_hybrid(hybrid, nt, dt, hvcoord, elem, nets, nete,
 #endif
 
        do qi = 1,qsize
-          if (q_adjustment) then
-             ! FV Q_ten
-             !   GLL Q0 -> FV Q0
-             call gfr_g2f_mixing_ratio(gfr, ie, elem(ie)%metdet, dp, dp_fv, &
-                  elem(ie)%state%Q(:,:,:,qi), wf1)
-             !   FV Q_ten = FV Q1 - FV Q0
-             wf1(:nf2,:) = q(:nf2,:,qi,ie) - wf1(:nf2,:)
-             if (nf > 1 .or. .not. gfr%boost_pg1) then
-                ! GLL Q_ten
-                call gfr_f2g_scalar_dp(gfr, ie, elem(ie)%metdet, dp_fv, dp, wf1, wg1)
-                ! GLL Q1
-                elem(ie)%derived%FQ(:,:,:,qi) = elem(ie)%state%Q(:,:,:,qi) + wg1
-             else
-                ! GLL Q_ten
-                do k = 1,nlev
-                   elem(ie)%derived%FQ(:,:,k,qi) = wf1(1,k)
-                end do
-             end if
-             ! Get limiter bounds.
-             do k = 1,nlev
-                gfr%qmin(k,qi,ie) = minval(q(:nf2,k,qi,ie))
-                gfr%qmax(k,qi,ie) = maxval(q(:nf2,k,qi,ie))
-             end do
+          ! FV Q_ten
```

Comment:
So the fv->gll remap routine now does the same operation, regardless of ftype? Namely, it seems that now you do `dQ_fv = Q_new_fv - g2f(Q_old_gll)`, and then set `FQ=Q_old_gll + f2g(dQ_fv)` (at least for pg2 case). So FQ always contains a tracers value. But looking at applyCAMforcing_tracers, for ftype=0, FQ should contain a tendency. What am I missing?

Reply:
Right, it turns out gllfvremap_mod doesn't need to know ftype. stepon.F90 does the conversion from state to tendency, starting at this line: https://github.com/E3SM-Project/E3SM/blob/3a87fa1738e1bfb91fb84acbf2cadbf88dddcef1/components/eam/src/dynamics/se/stepon.F90#L383

Reply:
Ah, gotcha, thanks. This makes sense.